pkg/processor/runtime/nodejs/test/outputter/outputter.js

```diff
@@ -15,60 +15,60 @@ limitations under the License.
-	return;
+        return;
     }
 
-    var body = event.body.toString();
+    const body = event.body.toString();
 
     switch (body) {
-	case 'return_string':
-	    context.callback('a string');
-	    return;
-	case 'return_status_and_string':
-	    context.callback([201, 'a string after status']);
-	    return;
-	case 'return_list':
-	    context.callback([{a: 1}, {b: 2}]);
-	    return;
-	case 'return_status_and_dict':
-	    context.callback([201, {a: 'dict after status', b: 'foo'}]);
-	    return;
-	case 'log':
-	    context.logger.debug('Debug message');
-	    context.logger.info('Info message');
-	    context.logger.warn('Warn message');
-	    context.logger.error('Error message');
-	    context.callback([201, 'returned logs']);
-	    return;
-	case 'log_with':
-	    context.logger.errorWith('Error message', {source: 'rabbit', weight: 7});
-	    context.callback([201, 'returned logs with']);
-	    return;
-	case 'return_response':
-	    headers = event.headers;
-	    headers['h1'] = 'v1';
-	    headers['h2'] = 'v2';
+        case 'return_string':
+            context.callback('a string');
+            return;
+        case 'return_status_and_string':
+            context.callback([201, 'a string after status']);
+            return;
+        case 'return_list':
+            context.callback([{a: 1}, {b: 2}]);
+            return;
+        case 'return_status_and_dict':
+            context.callback([201, {a: 'dict after status', b: 'foo'}]);
+            return;
+        case 'log':
+            context.logger.debug('Debug message');
+            context.logger.info('Info message');
+            context.logger.warn('Warn message');
+            context.logger.error('Error message');
+            context.callback([201, 'returned logs']);
+            return;
+        case 'log_with':
+            context.logger.errorWith('Error message', {source: 'rabbit', weight: 7});
+            context.callback([201, 'returned logs with']);
+            return;
+        case 'return_response':
+            let headers = event.headers;
+            headers['h1'] = 'v1';
+            headers['h2'] = 'v2';
 
-	    context.callback(new context.Response('response body', headers, 'text/plain', 201));
-	    return;
-	case 'return_fields':
-	    var fields = [];
-	    for (var key in event.fields) {
-		fields.push(key + '=' + event.fields[key]);
-	    }
-	    // We use sorted to get predictable output
-	    fields.sort();
-	    context.callback(fields.join(','));
-	    return;
-	case 'return_path':
-	    context.callback(event.path);
-	    return;
-	case 'return_error':
-	    throw 'some error';
-	default:
-	    throw 'Unknown return mode: ' + event.body;
+            context.callback(new context.Response('response body', headers, 'text/plain', 201));
+            return;
+        case 'return_fields':
+            let fields = [];
+            for (const key in event.fields) {
+                fields.push(key + '=' + event.fields[key]);
+            }
```

Comment:
Okay after all above discussion about `for...of` loop, I see that you don't need even _that_.

What you do here is initialize then push something from another array:
```js
const source = [1, 2, 3, 4]
const destination = []
for (const value of source) {
    destination.push(value + 1)
}
// now destination is [2, 3, 4, 5]
```
This is a classic case for a mapping function, so you have a single statement, `destination` array is initialized to a non-empty array that already has all the values you want:
```js
const source = [1, 2, 3, 4]
const destination = source.map(value => value + 1)
// now destination is [2, 3, 4, 5]
```
If you don't want to use arrow function (though there's no reason not to do it here):
```js
const source = [1, 2, 3, 4]
const destination = source.map(function (value) {
    return value + 1
})
// now destination is [2, 3, 4, 5]
```
In our case:
```js
const fields = event.fields.map((value, key) => `${key}=${value}`)
```

Reply:
🚀